webplugin/js/app/zendesk-chat-service.js

```diff
@@ -166,6 +169,16 @@ function ZendeskChatService() {
         }
     };
 
+    _this.getBotNameById = function(botId, userDetails) {
+        for (var i = 0; i<userDetails.length; i++) {
+            var currentUserDetail = userDetails[i];
+            if (currentUserDetail.userId == botId) {
+                return currentUserDetail.userName;
+            }
+        }
```

Comment:
1. can be achieved by simple .find().
2. does not seem efficient, can you think of some more optimised approach

Reply:
Didn't use find because its not supported on IE.  Need to think of something else then👍
<img width="927" alt="image" src="https://user-images.githubusercontent.com/36797892/161979579-7e34324f-75fd-47ec-ac60-fafcc133fe93.png">

Reply:
we maintain some map also MCK_USER_DETAIL_MAP or something, see if that can be used

Reply:
What I have found is this `userDetails` array generally consists of 4-5 users.

1. The user that is using the chat-widget.
2. The assigned bot for the conversation.
3. The assigned Agent after handoff.
4. And this `suraj++13@applozic.com` is present in every group on that particular account. I guess because this email is added as an Admin or something.

<img width="563" alt="image" src="https://user-images.githubusercontent.com/36797892/162370073-628e25b5-ed22-4131-83c8-ef088e37e682.png">

So this should not cause performance issues. @devashishmamgain @gargshantnu